Snippets/Snippets_6/Lifecycle/EndpointStartAndStop.cs

```diff
@@ -2,19 +2,48 @@
 {
     using System.Threading.Tasks;
     using NServiceBus;
+    using NServiceBus.Features;
 
     #region lifecycle-EndpointStartAndStop
 
-    class RunWhenEndpointStartsAndStops : IWantToRunWhenEndpointStartsAndStops
+    class RunWhenEndpointStartsAndStops
     {
-        public async Task Start(IMessageSession session)
+        static async Task StartAndStopEndpoint()
         {
+            EndpointConfiguration endpointConfiguration = new EndpointConfiguration("SelfHosted");
+
+            IEndpointInstance endpoint = await Endpoint.Start(endpointConfiguration);
+
             // perform startup logic
+
+            await endpoint.Stop();
+
+            // perform shutdown logic
         }
+    }
 
-        public async Task Stop(IMessageSession session)
+    class RunWhenEndpointStartsAndStopsFeature : Feature
```

Comment:
I couldn't remember where we landed on including this, so I put it here anyhow. The only reason for including it is for users that are using our container, which we are discouraging.